versions.sh

```diff
@@ -1,9 +1,17 @@
 #!/bin/bash
 set -Eeuo pipefail
 
+# the libc created by gcc might be too old for a newer Debian:
+# https://packages.debian.org/stable/gcc
+# bookworm: 12.2
+# bullseye: 10.2
+# buster: 8.3
```

Comment:
Note for posterity: yes, this means that `gcc:9` should _technically_ not be updated to `bullseye`, but it's been on `bullseye` for a while without any obvious complaints so I'd be hesitant to downgrade it now :sweat_smile:

Reply:
And the inline test added to the `RUN` shows that it provides a new enough `GLIBC*` compatibility.